### ISSUE TITLE
Redis endpoint null reference in CAP.OpenTelemetry

### ISSUE THREAD

**Issue by jakey188** [dotnetcore/CAP#1382]
报错堆信息: 在 DotNetCore.CAP.OpenTelemetry.DiagnosticListener.OnNext(KeyValuePair`2 evt)

麻烦帮忙看看是哪里的问题

NET框架：net6

CAP版本:
DotNetCore.CAP 7.2.0
DotNetCore.CAP.Dashboard 7.2.0
DotNetCore.CAP.MySql 7.2.0
DotNetCore.CAP.OpenTelemetry 7.2.0
DotNetCore.CAP.RedisStreams 7.2.0

OpenTelemetry版本:
OpenTelemetry.Exporter.Console 1.5.1
OpenTelemetry.Extensions.Hosting 1.5.1
OpenTelemetry.Instrumentation.AspNetCore 1.5.1-beta.1

Program代码如下：

```
builder.Services.AddCap(cap =>
{
    cap.UseMySql(c =>
    {
        c.ConnectionString = "server=localhost;Database=test11;Uid=root;Pwd=123456;charset=utf8";
    });
    cap.UseRedis(c =>
    {
        c.Configuration = ConfigurationOptions.Parse("127.0.0.1");
        c.StreamEntriesCount = 10; //读取时从 stream 返回的条目数,默认10
        c.ConnectionPoolSize = 10; //连接池数,默认10
    });
    cap.UseDashboard();
});
//.AddSubscribeFilter<CapTestFilter>();

builder.Services.AddOpenTelemetry()
        .WithTracing(builder => builder
            .AddAspNetCoreInstrumentation()
            .AddCapInstrumentation()    // <-- 添加这行
            .AddConsoleExporter());

var app = builder.Build();
```
Controller：
 ```
[HttpGet("p1")]
        public async Task<IActionResult> p1()
        {
            var user = new User { id = 1, name = "hello" };

            await _publisher.PublishAsync<User>("cap.test111", user);

            return Ok();
        }

        [NonAction]
        [CapSubscribe("cap.test111")]
        public void Sub1(User user)
        {
            Console.WriteLine(user?.name);
        }
```

**Comment by yang-xiaodong** [comment]
Hi @jakey188  

I'm busy recently, Would you like to investigate this issue?

**Comment by jakey188** [comment]
经过调试发现了问题所在

`public BrokerAddress BrokerAddress => new ("redis", _options.Endpoint);`

```
activity.SetTag("messaging.system", eventData.BrokerAddress.Name);
activity.SetTag("messaging.destination", eventData.Operation);
activity.SetTag("messaging.destination_kind", "topic");
activity.SetTag("messaging.url", eventData.BrokerAddress.Endpoint!.Replace("-1", "5672"));
activity.SetTag("messaging.message_id", eventData.TransportMessage.GetId());
activity.SetTag("messaging.message_payload_size_bytes", eventData.TransportMessage.Body.Length);
```

activity.SetTag("messaging.url", eventData.BrokerAddress.Endpoint!.Replace("-1", "5672"));

eventData.BrokerAddress.Endpoint! 未赋值,这里报错。

**Comment by jakey188** [comment]
我尝试提交一个pr解决这个bug

**Comment by yang-xiaodong** [comment]
Fixed in [7.2.1](https://github.com/dotnetcore/CAP/releases/tag/v7.2.1)
